action/protocol/staking/delegate_map.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2020 IoTeX Foundation
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package staking
+
+import (
+	"sort"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/pkg/errors"
+
+	"github.com/iotexproject/iotex-core/action/protocol/staking/stakingpb"
+)
+
+type (
+	// DelegateMap is a map of delegates using owner's 12-byte name as key
+	DelegateMap map[CandName]*Delegate
```

Comment:
I think you would need two maps. one is CandName to OwnerAddress mapping and one is OwnerAddress to Delegate instance mapping